### PR TITLE
RUM-6235 Handle sse request

### LINKED ISSUE #2266: Okhttp with DatadogInterceptor is blocking SSE streams
### Describe the bug

Hi 

The app I'm working on has a DatadogInterceptor and it's working normally until I implemented the Server-Sent Events.
Even though the streams are coming in between certain interval, app receives all the streams at once.

But when I remove the `DatadogInterceptor`, the app receives the streams normally.

Need your help to fix it.

### Reproduction steps

Add datadog interceptor to okhttpclient:
```
List<String> tracedHosts = List.of(BuildConfig.BFF_BASE_URL);
okHttpClientBuilder.addInterceptor(new DatadogInterceptor.Builder(tracedHosts).build())
```
okHttpClientBuilder is passed to create httpclient engine.

Create an SSE endpoint that sends data
```
    sse(OrderStatusResource.PATH) {

        val sseFlow = flow {
            statuses.forEach {
                println(it)
                emit(SseEvent(data = it))
                delay(5.seconds)
            }
        }.shareIn(GlobalScope, SharingStarted.WhileSubscribed(5_000))

        sseFlow.collect {
            val json = Json { encodeDefaults = true }
            val data = json.encodeToString(it)
            send(ServerSentEvent(data = data))
        }
    }
```

now consume the data on the client
```
        httpClient.sse(host = "192.168.251.226", port = 8080, path = OrderStatusResource.PATH) {
            incoming.collect { event ->
                println("SSE::>> $event")
                _flow.emit(OrderStatusDto(data = event.data.toString()))
                if (event.data.equals("Done")) {
                    httpClient.close()
                }
            }
        }
```

you will get all the payload at once when the connection closes.

but when i comment the `DatadogInterceptor`, everything works normally.

### Logcat logs

_No response_

### Expected behavior

App should receive SSE events normally

### Affected SDK versions

datadog-android 2.13.1

### Latest working SDK version

n/a

### Did you confirm if the latest SDK version fixes the bug?

Yes

### Kotlin / Java version

kotlin 2.0.0

### Gradle / AGP version

agp 8.6.0-rc1

### Other dependencies versions

com.squareup.okhttp3 = "4.12.0"
io.ktor:ktor-client-okhttp = "3.0.0-rc-1"

### Device Information

_No response_

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

This PR Handles properly SSE (Server Sent Event) network request instrumented with our OkHttp interceptor. 
In order to report the size of a downloaded resource, we use the `peekBody()` method which tries to read the whole content of the network request. In case of a SSE request, it would wait until the end of the stream to eventually complete the interception and forward the content to the receiver, blocking the server messages. 

ServerSent events should have the `text/server-event` Mime Type, which we can use to detect such use case. 

### Motivation

Fixes #2266

### CODE REVIEW

File: integrations/dd-sdk-android-okhttp/src/main/kotlin/com/datadog/android/okhttp/DatadogInterceptor.kt [hunk 1]
```diff
@@ -357,13 +357,19 @@ internal constructor(
 
     private fun getBodyLength(response: Response, internalLogger: InternalLogger): Long? {
         return try {
-            val body = response.body ?: return null
+            val body = response.body
+            val contentType = body?.contentType()?.let {
+                // manually rebuild the mimetype has `toString()` can also include the charsets
```
Comment (0xnm) [inline integrations/dd-sdk-android-okhttp/src/main/kotlin/com/datadog/android/okhttp/DatadogInterceptor.kt:362]:
```suggestion
                // manually rebuild the mimetype, `toString()` can also include the charsets
```

File: integrations/dd-sdk-android-okhttp/src/main/kotlin/com/datadog/android/okhttp/DatadogInterceptor.kt [hunk 2]
```diff
@@ -481,6 +487,10 @@ internal constructor(
 
     internal companion object {
 
+        internal val STREAM_CONTENT_TYPES = arrayOf(
+            "text/event-stream"
```
Comment (0xnm) [inline integrations/dd-sdk-android-okhttp/src/main/kotlin/com/datadog/android/okhttp/DatadogInterceptor.kt:491]: should we maybe try to include other stream types here? like for websocket, gRPC

File: sample/kotlin/build.gradle.kts [hunk 3]
```diff
@@ -195,6 +195,7 @@ dependencies {
     implementation("io.opentracing.contrib:opentracing-rxjava-3:0.1.4") {
         exclude(group = "io.opentracing")
     }
+    implementation("com.launchdarkly:okhttp-eventsource:2.5.0")
```
Comment (0xnm) [inline sample/kotlin/build.gradle.kts:198]: it is already included in the `// Network` section below

File: sample/kotlin/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -60,6 +60,7 @@
     <string name="button_async_operation">Async Operation</string>
     <string name="button_network_request">Network Request</string>
     <string name="button_start">Start</string>
+    <string name="button_start_sse">Start SSE Request</string>
```
Comment (0xnm) [inline sample/kotlin/src/main/res/values/strings.xml:63]: minor: maybe use full form? `Start Server-Side Event Request`

**Review by 0xnm — APPROVED** [review]

File: integrations/dd-sdk-android-okhttp/src/main/kotlin/com/datadog/android/okhttp/DatadogInterceptor.kt (hunk 1 above)
Comment (xgouchet) [reply integrations/dd-sdk-android-okhttp/src/main/kotlin/com/datadog/android/okhttp/DatadogInterceptor.kt]: 👍

File: sample/kotlin/build.gradle.kts (hunk 3 above)
Comment (xgouchet) [reply sample/kotlin/build.gradle.kts]: 👍

File: sample/kotlin/src/main/res/values/strings.xml (hunk 4 above)
Comment (xgouchet) [reply sample/kotlin/src/main/res/values/strings.xml]: I could but it'll make the button multiline on most device 😅

File: integrations/dd-sdk-android-okhttp/src/main/kotlin/com/datadog/android/okhttp/DatadogInterceptor.kt [hunk 5]
```diff
@@ -481,6 +487,10 @@ internal constructor(
 
     internal companion object {
 
+        internal val STREAM_CONTENT_TYPES = arrayOf(
```
Comment (0xnm) [inline integrations/dd-sdk-android-okhttp/src/main/kotlin/com/datadog/android/okhttp/DatadogInterceptor.kt:490]: should it be `setOf`?

File: integrations/dd-sdk-android-okhttp/src/main/kotlin/com/datadog/android/okhttp/DatadogInterceptor.kt (hunk 2 above)
Comment (xgouchet) [reply integrations/dd-sdk-android-okhttp/src/main/kotlin/com/datadog/android/okhttp/DatadogInterceptor.kt]: Good idea

**Review by 0xnm — APPROVED** [review]
there is one comment about `arrayOf` -> `setOf` (performance-wise on such small collection it won't matter though), otherwise lgtm

File: gradle/libs.versions.toml [hunk 6]
```diff
@@ -89,7 +89,8 @@ timber = "5.0.1"
 coroutines = "1.4.2"
 
 # Local Server
-ktor = "1.6.0"
+ktor = "1.6.8"
+ktorServer = "3.0.0-rc-1"
```
Comment (mariusc83) [inline gradle/libs.versions.toml:93]: is there a concern if we are using a `rc` here ?

File: integrations/dd-sdk-android-okhttp/src/main/kotlin/com/datadog/android/okhttp/DatadogInterceptor.kt (hunk 5 above)
Comment (xgouchet) [reply integrations/dd-sdk-android-okhttp/src/main/kotlin/com/datadog/android/okhttp/DatadogInterceptor.kt]: 👍

File: gradle/libs.versions.toml (hunk 6 above)
Comment (xgouchet) [reply gradle/libs.versions.toml:93]: This lib is only used for the sample app, so it's ok

**Review by mariusc83 — APPROVED** [review]

**Pull request merged**